pkg/console/service/kubevirt-proxy_test.go

```diff
@@ -1,9 +1,221 @@
+			})
+
+			apiClient.Fake.PrependReactor("create", "subjectaccessreviews", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				createAction := action.(k8stesting.CreateAction)
+				sar := createAction.GetObject().(*authzv1.SubjectAccessReview)
+				sar.Status.Allowed = true
+				return true, sar, nil
+			})
+
+			apiClient.Fake.PrependReactor("create", "serviceaccounts/token", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				createAction := action.(k8stesting.CreateAction)
+				tokenRequest := createAction.GetObject().(*authnv1.TokenRequest)
+				tokenRequest.Status.Token = testToken
+				return true, tokenRequest, nil
+			})
+
+			request.Request.Header.Set(authorizationHeader, "Bearer "+validToken)
+			// Using a dummy URL, so tests don't panic
+			requestUrl, err := url.Parse("example.org/api")
+			Expect(err).ToNot(HaveOccurred())
+			request.Request.URL = requestUrl
+		})
+
+		It("should fail if VM does not exist", func() {
+			testVm = nil
+
+			testService.TokenHandler(request, response)
+
+			Expect(recorder.Code).To(Equal(http.StatusNotFound))
+			Expect(recorder.Body.String()).To(ContainSubstring("VirtualMachine does not exist"))
+		})
+
+		It("should return token", func() {
+			testService.TokenHandler(request, response)
+
+			Expect(recorder.Code).To(Equal(http.StatusOK))
+
+			tokenResponse := &api.TokenResponse{}
+			Expect(json.NewDecoder(recorder.Body).Decode(tokenResponse)).To(Succeed())
+
+			Expect(tokenResponse.Token).To(Equal(testToken))
+		})
+
+		It("should fail if duration parameter fails to parse", func() {
+			urlWithDuration, err := url.Parse("example.org/api?duration=this-fails-to-parse")
+			Expect(err).ToNot(HaveOccurred())
+			request.Request.URL = urlWithDuration
+
+			testService.TokenHandler(request, response)
+
+			Expect(recorder.Code).To(Equal(http.StatusBadRequest))
+			Expect(recorder.Body.String()).To(ContainSubstring("failed to parse duration"))
+		})
+
+		It("should return token with specified duration", func() {
+			apiClient.Fake.PrependReactor("create", "serviceaccounts/token", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				createAction := action.(k8stesting.CreateAction)
+				tokenRequest := createAction.GetObject().(*authnv1.TokenRequest)
+				Expect(*tokenRequest.Spec.ExpirationSeconds).To(Equal(int64(24 * 3600)))
+
+				tokenRequest.Status.Token = testToken
+				return true, tokenRequest, nil
+			})
+
+			urlWithDuration, err := url.Parse("example.org/api?duration=24h")
+			Expect(err).ToNot(HaveOccurred())
+			request.Request.URL = urlWithDuration
+
+			testService.TokenHandler(request, response)
+
+			Expect(recorder.Code).To(Equal(http.StatusOK))
+
+			tokenResponse := &api.TokenResponse{}
+			Expect(json.NewDecoder(recorder.Body).Decode(tokenResponse)).To(Succeed())
+			Expect(tokenResponse.Token).To(Equal(testToken))
+		})
+
+		It("should create resources", func() {
+			const resourceName = testName + "-vnc-access"
```

Comment:
Maybe we can generate `testName + "-vnc-access"` from a function?
```
func generateResourceName(name string) {
return name  + "-vnc-access"
}
```

Reply:
No, I don't want to call private function from unit tests, and exporting that function just for unit test is unnecessary.